backend/dal/Services/Concrete/ProjectReportService.cs

```diff
@@ -184,19 +184,19 @@ public ProjectReport Update(ProjectReport report)
             var originalReport = this.Context.ProjectReports
                 .SingleOrDefault(p => p.Id == report.Id) ?? throw new KeyNotFoundException();
 
-            if (originalReport.To == null) throw new ArgumentNullException();
+            if (report.To == null) throw new ArgumentNullException();
```

Comment:
these values don't make sense to check the originals.

Reply:
Should always include the parameter that is null in the exception `nameof(report))`